shared/gh/js/views/gh.admin-batch-edit-date.js

```diff
@@ -107,46 +115,55 @@ define(['lodash', 'moment', 'gh.core', 'gh.api.config'], function(_, moment, gh,
-                    eventDay = dateByWeekAndDay.getDate();
-                    var eventStartHour = parseInt($('#gh-batch-edit-hours-start', $timePickerContainer).val(), 10);
-                    var eventStartMinute = parseInt($('#gh-batch-edit-minutes-start', $timePickerContainer).val(), 10);
-                    var eventEndHour = parseInt($('#gh-batch-edit-hours-end', $timePickerContainer).val(), 10);
-                    var eventEndMinute = parseInt($('#gh-batch-edit-minutes-end', $timePickerContainer).val(), 10);
-
-                    // Create the start date of the event
-                    var startDate = moment([eventYear, eventMonth, eventDay, eventStartHour, eventStartMinute, 0, 0]);
-                    // Create the end date of the event
-                    var endDate = moment([eventYear, eventMonth, eventDay, eventEndHour, eventEndMinute, 0, 0]);
-
-                    // Send off an event that will be picked up by the batch edit and add the rows to the terms
-                    var alreadyAdded = $('.info .gh-event-date[data-start="' + gh.utils.convertUnixDatetoISODate(startDate.utc().format()) + '"]').length;
-                    if (!alreadyAdded) {
-                        $(document).trigger('gh.batchedit.addevent', {
-                            'eventContainer': $('.gh-batch-edit-events-container[data-term="' + termName + '"]').find('tbody'),
-                            'eventObj': {
-                                'tempId': gh.utils.generateRandomString(), // The actual ID hasn't been generated yet
-                                'isNew': true, // Used in the template to know this one needs special handling
-                                'selected': true,
-                                'displayName': $('.gh-jeditable-series-title').text(),
-                                'end': gh.utils.convertUnixDatetoISODate(moment(endDate).utc().format()),
-                                'location': '',
-                                'notes': 'Lecture',
-                                'organisers': null,
-                                'start': gh.utils.convertUnixDatetoISODate(moment(startDate).utc().format())
-                            }
-                        });
-                    }
-                });
+                var $days = $('.gh-batch-edit-time-picker');
+                if ($days.length) {
+                    _.each($days, function($timePickerContainer) {
+                        // Get the week number
+                        var eventWeek = parseInt(chk.value, 10);
+                        // Get the day number
+                        var eventDay = parseInt($('#gh-batch-edit-day-picker', $timePickerContainer).val(), 10);
+                        // Get the date by week and day
+                        var dateByWeekAndDay = gh.utils.getDateByWeekAndDay(termName, eventWeek, eventDay);
+
+                        var eventYear = dateByWeekAndDay.getFullYear();
```

Comment:
Currently the following error gets thrown: `Uncaught TypeError: Cannot read property 'getFullYear' of null` when using the batch time edit.